PowerPoint_Adapter/CRUD/Update/Table.cs

```diff
@@ -0,0 +1,248 @@
+                case GraphicFrame graphicFrame:
+                    frame = graphicFrame;
+                    break;
+                case Shape shape:
+                    frame = ConvertToGraphicFrame(shape, rowCount, columnCount);
+                    slidePart.Slide.CommonSlideData.ShapeTree.ReplaceChild(frame, shape);
+                    break;
+                default:
+                    BH.Engine.Base.Compute.RecordError($"The element with name '{update.ElementName}' on slide {update.SlideNumber} must be either a Shape, a Table, or a Table placeholder to be updated with a table.");
+                    return;
+            }
+
+            D.Table table = frame.Graphic?.GraphicData?.GetFirstChild<D.Table>();
+
+            if (table == null)
+            {
+                BH.Engine.Base.Compute.RecordError($"The element with the name `{update.ElementName}` on slide {update.SlideNumber} must contain a Table.");
+                return;
+            }
+
+            IEnumerable<D.GridColumn> columns = table.TableGrid.Descendants<D.GridColumn>();
+            IEnumerable<D.TableRow> rows = table.Descendants<D.TableRow>();
+
+            //compare to the column and row counts in the data to update.
+            if (columns.Count() != columnCount || rows.Count() != rowCount)
+            {
+                BH.Engine.Base.Compute.RecordError($"The shape of the data in Content must be the same shape as the table that is being updated.\nContent has {rowCount} rows and {columnCount} columns, whereas the table has {rows.Count()} rows and {columns.Count()} columns.");
+                return;
+            }
+
+            //update the text in each row/column - This assumes that each table cell will have only one paragraph and run originally and does not modify any other properties.
+
+            int r = 0;
+            int c = 0;
+            bool headerRow = update.HeaderRow.Count != 0;
+            foreach (D.TableRow row in rows)
+            {
+                foreach (D.TableCell cell in row.Descendants<D.TableCell>())
+                {
+                    try
+                    {
+                        D.Paragraph par = cell.TextBody.Elements<D.Paragraph>().SingleOrDefault();
+                        if (par == null)
+                        {
+                            par = new D.Paragraph();
+                            par.AddChild(new D.Run(new D.RunProperties(), new D.Text()));
+                            cell.TextBody.AddChild(par);
+                        }
+
+                        D.Run run = par.Elements<D.Run>().SingleOrDefault();
+                        int? fontSize = null;
+                        if (run == null)
+                        {
+                            run = new D.Run(new D.RunProperties(), new D.Text());
+                            par.AddChild(run);
+                        }
+                        else
+                            fontSize = run.RunProperties.FontSize / 100;
```

Comment:
This fails when `FontSize` is null. Which is the case for the table placeholder.